
# Result files
The pipeline line generates lots of files of which only some are kept. These final output files are listed and shortly described in the table below.
{sample} is replaced with the sample name and {type} is replaced with the sample type (T/N/R, where T=Tumor, N=Normal, R=RNA). The most important files and shown in bold.

## Main pipeline output files
<br />

| **File** | **File type** | **Description** |
|-|-|-|
| **Alignment** | :~~: | ~~ |
| **`bam_dna/{sample}_{type}.bam`** | bam | Alignment file |
| `bam_dna/{sample}_{type}.bam.bai` | bai | Alignment index file |
| **`bam_rna/{sample}_{type}.star_fusion.bam`** | bam | Alignment file  |
| `bam_rna/{sample}_{type}.star_fusion.bam.bai` | bai | Alignment index file |
| `bam_dna/mutect2_indel_bam/{sample}_{type}.bam` | bam | Realigned regions around INDELs |
| `bam_dna/mutect2_indel_bam/{sample}_{type}.bam.bai` | bai | Alignment index file |
| **SNV and INDELs** | :~~: | ~~ |
| **`results/dna/vcf/`<br />`{sample}_{type}.annotated.exon_only.filter.hard_filter.codon_snv.vcf`** | vcf | Final hard filtered and annotated file with SNV and INDEL variants. <br /> **OBS! Do not use as input to QCI** |
| **`results/dna/vcf/`<br />`{sample}_{type}.annotated.exon_only.filter.hard_filter.codon_snv.qci.vcf`** | vcf | Final hard filtered and annotated file with SNV and INDEL variants <br />used as input to QCI |
| `gvcf_dna/{sample}_{type}.mosdepth.g.vcf.gz` | g.vcf | Genomic vcf file with coverage in all position in the design |
| **CNVs** | :~~: | ~~ |
|  **`results/dna/cnv/{sample}_{type}/{sample}_{type}.pathology.cnv.html`** | html | Interactive html report of the filtered CNV results <br />using tumor content estimated by a pathologist |
|  `results/dna/cnv/{sample}_{type}/{sample}_{type}.purecn.cnv.html` | html | Interactive html report of the filtered CNV results <br />using tumor content estimated by PureCN |
|  **`results/dna/cnv/{sample}_{type}/{sample}_{type}.pathology.cnv_report.tsv`** | table | Table of filtered CNVs using tumor content <br />estimated by a pathologist |
|  `results/dna/cnv/{sample}_{type}/{sample}_{type}.purecn.cnv_report.tsv` | table | Table of filtered CNVs using tumor content <br />estimated by PureCN |
|  `results/dna/cnv/{sample}_{type}/{sample}_{type}.deletions.tsv` | table | Table of called small CNVs by in-house script |
|  `results/dna/cnv/{sample}_{type}/{sample}_{type}.purecn.purity.txt` | table | Estimated tumor content and ploidity by PureCN |
| **Fusions** | :~~: | ~~ |
|  **`results/dna/fusion/{sample}_{type}.gene_fuse_report.tsv`** | table | Filtered report of DNA fusions from GeneFuse |
|  **`results/rna/fusion/{sample}_{type}.fusion_report.tsv`** | table | Filtered report of RNA fusions from three different callers |
|  **`results/rna/fusion/{sample}_{type}.exon_skipping.tsv`** | table | Called exon skipping in MET and EGFR |
|  `results/rna/fusion/{sample}_{type}.arriba.fusions.pdf` | image | Image generated by Arriba visualizing the called fusions |
| **Biomarker** | :~~: | ~~ |
|  **`results/dna/msi/{sample}_{type}.msisensor_pro.score.tsv`** | table | MSI score in the form percent of sites that show instability |
|  **`results/dna/tmb/{sample}_{type}.TMB.txt`** | table | TMB score and SNV variants used to calculate the score. <br />**OBS! use the nsSNV TMB score** |
|  `results/dna/hrd/{sample}_{type}.pathology.scarhrd_cnvkit_score.txt` | table | HRD score using Tumor content estimated by a pathologist <br />**OBS! experimental** |
|  `results/dna/hrd/{sample}_{type}.purecn.scarhrd_cnvkit_score.txt` | table | HRD score using Tumor content estimated by PureCN <br />**OBS! experimental** |
| **QC** | :~~: | ~~ |
|  **`results/dna/qc/multiqc_DNA.html`** | html | MultiQC report for DNA samples |
|  **`results/rna/qc/multiqc_RNA.html`** | html | MultiQC report for RNA samples |
|  `results/dna/qc/{sample}_{type}.coverage_and_mutations.tsv` | table | Report with coverage in clinical important positions as well as <br />additional variant information |
|  `results/rna/qc/{sample}_{type}.house_keeping_gene_coverage.tsv` | table | Median coverage in four house keeping genes |
|  `results/rna/id_snps/{sample}_{type}.id_snps.vcf` | vcf | SNV file of id SNPs also present in the DNA panel |

## Additional pipeline output files
Additional files saved mainly for troubleshooting and development are located under results/additional_files.  
<br />

| **File** | **File type** | **Description** |
|-|-|-|
| **SNV and INDELs** | :~~: | ~~ |
| `results/additional_files/dna/vcf/{caller}_{sample}_{type}.vcf.gz` | vcf | SNV and INDEL variants file from the individual callers |
| `results/additional_files/dna/vcf/{sample}_{type}.annotated.vcf.gz` | vcf | Merged and annotated SNV and INDEL variants file |
| `results/additional_files/dna/vcf/`<br />`{sample}_{type}.annotated.exon_only.filter.soft_filter.vcf` | vcf | Soft filtered, merged and annotated SNV and INDEL variants file <br />without double variants |
| `results/additional_files/dna/vcf/`<br />`{sample}_{type}.annotated.exon_only.filter.hard_filter.vcf` | vcf | Hard filtered, merged and annotated SNV and INDEL variants file <br />without double variants |
| **CNVs** | :~~: | ~~ |
|  `results/additional_files/dna/cnv/{sample}_{type}/{sample}_{type}.manta_tumorSV.vcf.gz` | vcf | Manta variant calling |
|  `results/additional_files/dna/cnv/{sample}_{type}/{sample}_{type}.cnvkit.scatter.png` | image | CNVkit genome CNV plot |
|  `results/additional_files/dna/cnv/{sample}_{type}/{sample}_{type}.cnvkit.diagram.pdf` | image | CNVkit chromosome CNV plot |
|  `results/additional_files/dna/cnv/{sample}_{type}/{sample}_{type}.purecn.svdb_query.vcf` | vcf | Merged CNV vcf by SVDB from the two callers <br />using tumor content estimated by a pathologist |
|  `results/additional_files/dna/cnv/{sample}_{type}/{sample}_{type}.pathology.svdb_query.vcf` | vcf | Merged CNV vcf by SVDB from the two callers <br />using tumor content estimated by a PureCN |
| **Fusions** | :~~: | ~~ |
|  `results/additional_files/dna/fusion/{sample}_{type}.gene_fuse_fusions.txt` | table | Fusions called by GeneFuse |
|  `results/additional_files/rna/fusion/{sample}_{type}.arriba.fusions.tsv` | table | Fusions called by Arriba |
|  `results/additional_files/rna/fusion/{sample}_{type}.star-fusion.fusion_predictions.tsv` | table | Fusions called by Star-fusion |
|  `results/additional_files/rna/fusion/{sample}_{type}.fusioncatcher.fusion_predictions.txt` | table | Fusions called by FusionCatcher |
| **QC** | :~~: | ~~ |
|  `results/additional_files/dna/qc/{sample}_{type}.alignment_summary_metrics.txt` | table | Picard alignment stats |
|  `results/additional_files/dna/qc/{sample}_{type}.contamination.table` | table | GATK contamination stats |
|  `results/additional_files/dna/qc/{sample}_{type}.duplication_metrics.txt` | table | Picard sequence duplication stats |
|  `results/additional_files/dna/qc/{sample}_{type}.HsMetrics.txt` | table | Picard panel stats |
|  `results/additional_files/dna/qc/{sample}_{type}.insert_size_metrics.txt` | table | Picard insert size stats |
|  `results/additional_files/dna/qc/{sample}_{type}.samtools-stats.txt` | table | Samtools sequencing stats |
