# QC
See the [qc hydra-genetics module](https://snv_indels.readthedocs.io/en/latest/) documentation for more details on the softwares for the quality control. Default hydra-genetics settings/resources are used if no configuration is specfied.


## Pipeline output files:

* `results/rna/qc/multiqc_RNA.html`
* `results/rna/qc/{sample}_{type}.house_keeping_gene_coverage.tsv`

## MultiQC
A MultiQC html report is generated using **[MultiQC](https://github.com/ewels/MultiQC)** v1.11. The report starts with a general statistics table showing the most important QC-values followed by additional QC data and diagrams. The qc data is generated using FastQC, mosdepth, samtools, and picard.

### Configuration
**Software settings**

* `multiqc: reports: RNA: qc_files` - configuration of input files to MultiQC in the config file

```yaml
# config.yaml
multiqc:
  container: "docker://hydragenetics/multiqc:1.11"
  reports:
    RNA:
      config: "config/multiqc_config_rna.yaml"
      included_unit_types: ["R"]
      qc_files:
        - "qc/fastqc/{sample}_{type}_{flowcell}_{lane}_{barcode}_fastq1_fastqc.zip"
        - "qc/fastqc/{sample}_{type}_{flowcell}_{lane}_{barcode}_fastq2_fastqc.zip"
        - "qc/mosdepth/{sample}_{type}.mosdepth.global.dist.txt"
        - "qc/mosdepth/{sample}_{type}.mosdepth.region.dist.txt"
        - "qc/picard_collect_alignment_summary_metrics/{sample}_{type}.alignment_summary_metrics.txt"
        - "qc/picard_collect_hs_metrics/{sample}_{type}.HsMetrics.txt"
        - "qc/samtools_stats/{sample}_{type}.samtools-stats.txt"
        - "qc/mosdepth/{sample}_{type}.regions.bed.gz"
```

## FastQC
**[FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)** v0.11.9 is run on the raw fastq-files.

### Configuration
**Cluster resources**

| **Options** | **Value** |
|-------------|-|
| mem_mb | 12288 |
| mem_per_cpu | 6144 |
| threads | 2 |

## Samtools
**[Samtools stats](http://www.htslib.org/doc/samtools-stats.html)** v1.15 is run on Star aligned bam files generated by running Star-fusion.

## Picard
**[Picard](https://broadinstitute.github.io/picard/)** v2.25.0 is run on Star aligned bam files (from Star-fusion) collecting a number of metrics. The metrics calculated are listed below:

**picard CollectAlignmentSummaryMetrics** (using a fasta reference genome file)  
**picard CollectHsMetrics** (using a fasta reference genome file, a design bed file, and with the option COVERAGE_CAP=5000)

## Mosdepth
**[Mosdepth](https://github.com/brentp/mosdepth)** v0.3.2 is run on Star aligned bam files (from Star-fusion) together with a design bed file to gather coverage statistics both globally and locally per region.

### Configuration
**References**

* [RNA design bed](references.md#design_bed_rna)

<br />
**Software settings**

| **Options** | **Value** | **Description** |
|-------------|-|-|
| extra | --no-per-base | Do no output coverage per base |
| extra | --fast-mode | Fast coverage calculations | |

## House keeping gene coverage
House keeping gene coverage is reported by the in-house script [house_keeping_gene_coverage.py](https://github.com/genomic-medicine-sweden/Twist_Solid/blob/develop/workflow/scripts/house_keeping_gene_coverage.py) ([rule](https://github.com/genomic-medicine-sweden/Twist_Solid/blob/develop/workflow/rules/house_keeping_gene_coverage.smk)) that in turn uses **[Samtools depth](http://www.htslib.org/doc/samtools-depth.html)** to calculate the coverage. The RNA design bed file is used to define the regions that the coverage should be calculated in. The house keeping genes are listed below:

* House keeping genes:
    - GAPDH, GUSB, OAZ1, POLR2A

### Configuration
**Software settings for samtools (hard coded)**

| **Options** | **Value** | **Description** |
|-------------|-|-|
| -d | 5000000 | Max read depth |
| -a | |Report all positions even it is 0 coverage (for correct average calculations) |
| -r | {region} | Gene exon region to analyse |

<br />
**References**

* [RNA design bed](references.md#design_bed_rna)

### Result file

* `results/rna/qc/{sample}_{type}.house_keeping_gene_coverage.tsv`
